batchflow/models/torch/base_batchsize_mixin.py

```diff
@@ -0,0 +1,336 @@
+
+        # then run precise method in neighbourhood of batch_size_estimation
+        return self._compute_optimal_batch_size(inputs=inputs, targets=targets,
+                                                start_batch_size=batch_size_estimation,
+                                                low=low, high=high, method=method,
+                                                max_iters=max_iters, tail_size=tail_size,
+                                                max_memory=max_memory, pbar=pbar,
+                                                frequency=frequency, factor=factor,
+                                                update_method='binary')
+
+
+    def _bruteforce_batch_size_generator(self, factor, max_memory):
+        """ Calculates next batch size for bruteforce estimation method. If consumed memory is lower
+        than max_memory, then batch_size is multiplied by factor, otherwise it is divided by factor
+
+        Yields
+        ------
+        new_batch_size, exit: tuple(int, bool)
+            New batch size to check, and exit condition whether the optimal
+            batch size computation is finished
+        """
+
+        while True:
+            batch_size, consumed_memory = yield
+
+            if consumed_memory > max_memory:
+                yield batch_size // factor, True
+            else:
+                yield batch_size * factor, False
+
+
+    def _binary_batch_size_generator(self, low, high, max_memory):
+        """ Calculates next batch size for binary search method. If consumed memory is lower
+        than max_memory, then lower bound is increased, otherwise the upped bound is decreased.
+
+        Yields
+        ------
+        new_batch_size, exit: tuple(int, bool)
+            New batch size to check, and exit condition whether the optimal
+            batch size computation is finished.
+        """
+        while True:
+            batch_size, consumed_memory = yield
+
+            if consumed_memory > max_memory:
+                high = batch_size
+            else:
+                low = batch_size
+
+            exit = high - low <= 1
+            yield (high + low) // 2, exit
+
+
+    def _compute_optimal_batch_size(self, inputs=None, targets=None, low=2, high=None,
+                                    start_batch_size=4, max_iters=15, pbar='n',
+                                    method='train', max_memory=100, frequency=0.01,
+                                    tail_size=20, update_method='bruteforce',
+                                    factor=2):
+        count = 0
+        # if None => make equal distance between low, start_batch_size and high
+        high = high if high is not None else 2 * start_batch_size - low
+
+        # The list is used to show current batch_size in notifier. Current value is the last one.
+        batch_size = start_batch_size if isinstance(start_batch_size, list) else [start_batch_size]
```

Comment:
Maybe it would be better to separate these values into the two variables: `batch_size` and `batch_size_history`?

Reply:
It is better to use `list(start_batch_size)` instead of `[start_batch_size]`

Reply:
`start_batch_size` is an integer, so I can't use `list(start_batch_size)`, because it takes only iterable objects.